x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx

```diff
@@ -86,29 +86,38 @@ export function LatencyCorrelations({ onFilter }: { onFilter: () => void }) {
     startFetch,
     cancelFetch,
     overallHistogram,
-  } = useTransactionLatencyCorrelationsFetcher({
-    environment,
-    kuery,
-    serviceName,
-    transactionName,
-    transactionType,
-    start,
-    end,
-    percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
-  });
+  } = useTransactionLatencyCorrelationsFetcher();
+
+  const startFetchHandler = useCallback(() => {
+    startFetch({
+      environment,
+      kuery,
+      serviceName,
+      transactionName,
+      transactionType,
+      start,
+      end,
+      percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [kuery, start, end]);
```

Comment:
This should also depends on other parts of the params like `serviceName` or `environmentName`

Reply:
Fixed in 0762e556874